File: InsureX.Frontend/src/api/types/Asset.ts
import type { BaseEntity } from './Common';

export interface Asset extends BaseEntity {
    tenantId: number;
    assetType: 'Motor' | 'NonMotor';
    assetIdentifier: string;
    registrationNumber?: string;
    financedAmount: number;
    borrowerReference?: string;
    loanStartDate: string;
    loanEndDate: string;
    status: 'Active' | 'Settled' | 'Closed';
    complianceStatus?: string;
}

export interface Policy {
    id: number;
    policyNumber: string;
    insurerName: string;
    status: string;
    expiryDate: string;
    insuredValue: number;
}

export interface Borrower {
    name: string;
    idNumber: string;
    email: string;
    phone: string;
}

export interface ComplianceHistory {
    id: number;
    outcome: string;
    reason: string;
    evaluatedAt: string;
    correlationId: string;
}

export interface AssetDetail extends Asset {
    borrower: Borrower;
    policies: Policy[];
    complianceHistory: ComplianceHistory[];
}

export interface AssetFilterParams {
    page?: number;
    pageSize?: number;
    status?: string;
    assetType?: string;
    registrationNumber?: string;
}

File: IAPR_Data/Classes/Asset.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace IAPR_Data.Classes
{
    public class Asset
    {
        [Key]
        public int Id { get; set; }

        public int? TenantId { get; set; }

        [ForeignKey("TenantId")]
        public virtual Tenant? Tenant { get; set; }

        [Required]
        [StringLength(100)]
        public string AssetType { get; set; }

        [Required]
        [StringLength(100)]
        public string AssetIdentifier { get; set; }

        [StringLength(100)]
        public string RegistrationNumber { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal FinancedAmount { get; set; }

        [StringLength(200)]
        public string BorrowerReference { get; set; }

        public DateTime LoanStartDate { get; set; }
        public DateTime LoanEndDate { get; set; }

        [StringLength(50)]
        public string Status { get; set; }

        [StringLength(50)]
        public string ComplianceStatus { get; set; }

        public DateTime CreatedAt { get; set; }

        public Asset()
        {
            CreatedAt = DateTime.UtcNow;
            Status = "Active";
            ComplianceStatus = "Unknown";
        }
    }
}







